lib/apiservers/service/restapi/handlers/common.go

```diff
@@ -93,11 +93,10 @@ func buildDataAndValidateTarget(op trace.Operation, params buildDataParams, prin
 			return data, nil, util.NewError(http.StatusBadRequest, "Validation Error: datacenter parameter is not a datacenter moref")
 		}
 
-		data.Target.URL.Path = dc.InventoryPath
-
 		// Do what NewValidator would have done if a Datacenter had been set
 		v.DatacenterPath = dc.Name()
-		v.Session.DatastorePath = v.DatacenterPath
+		v.Session.DatacenterPath = v.DatacenterPath
```

Comment:
Since this is a nuanced fix, could we add a comment explaining why setting these two fields is critical?

Reply:
Good catch. Thanks!